src/Interpreters/MutationsInterpreter.h

```diff
@@ -82,19 +95,56 @@ class MutationsInterpreter
 
     void setApplyDeletedMask(bool apply) { apply_deleted_mask = apply; }
 
+    struct Source
```

Comment:
I don't like this class. The name is too common and collides with our basic `ISource` interface. Also, it's unclear to me which abstraction it builds. I would say, "it builds part of the mutation pipeline related to reading from storage," but it also adds some additional actions, specify settings, and so on. 

And also comment is missing.